### PR TITLE
Update to latest REST API

### PR DESCRIPTION
**Pull request by bgklein**

### CODE REVIEW

File: azext/batch/_file_utils.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def convert_blobs_to_resource_files(blobs, resource_properties):
         file_path = resource_properties.file_path if resource_properties.file_path \
             else blobs[0]['filePath']
         resource_files.append(models.ExtendedResourceFile(
-            blob_source=blobs[0]['url'],
+            http_url=blobs[0]['url'],
```
Comment (xingwu1) [inline azext/batch/_file_utils.py]: update the blobSource to httpUrl in template example and docs.

Comment (bgklein) [reply azext/batch/_file_utils.py]: I think you were looking at an older revision?

**Comment by bgklein** [comment]
@xingwu1 re-review

File: azext/batch/_file_utils.py [hunk 2]
```diff
@@ -320,16 +322,29 @@ def get_container_list(self, source):
 
     def resolve_resource_file(self, resource_file):
         """Convert new resourceFile reference to server-supported reference"""
-        if resource_file.blob_source:
+        if resource_file.http_url or resource_file.url:
```
Comment (xingwu1) [inline azext/batch/_file_utils.py]: what is the `url`?

File: azext/batch/models/__init__.py [hunk 3]
```diff
@@ -8,7 +8,6 @@
 # Not ideal syntax - but savaes us having to check and repopulate this
 # list every time the SDK is regenerated.
 from azure.batch.models import *
-from azure.batch.models.batch_service_client_enums import *
 
 try:
     from .extended_task_parameter_py3 import ExtendedTaskParameter
```
[the remark this answers is not in the document]
Comment (bgklein) [reply azext/batch/models/__init__.py]: We can we would just need to do it manually

**Comment by bgklein** [comment]
Force pushed to retrigger travis => last run was good minus copyright headers

File: azext/batch/models/constants.py [hunk 4]
```diff
@@ -63,3 +63,8 @@
 PROPS_ON_COLLECTION_TASK = PROPS_ON_REPEAT_TASK.union({
     'multiInstanceSettings',
     'dependsOn'})
+
+
+# Dates used as cutoffs for different SDK extension versions
+class KnownTemplateVersion(Enum):
+    Dec2018 = "2018-12-01"
```
Comment (xingwu1) [inline azext/batch/models/constants.py]: I think this is 2016-12-01?

File: azext/batch/operations/job_operations.py [hunk 5]
```diff
@@ -126,6 +130,12 @@ def add(self, job, job_add_options=None, custom_headers=None, raw=False,
          :class:`BatchErrorException<azure.batch.models.BatchErrorException>`
         """
         if isinstance(job, models.JobTemplate):
+            if job.api_version:
+                max_datetime = dt.strptime(KnownTemplateVersion.Dec2018, "%y-%m-%d")
+                specified_datetime = dt.strptime(job.api_version, "%y-%m-%d")
+                if max_datetime < specified_datetime:
+                    logger.error("The specified template API version is not supported by the current SDK extension")
```
Comment (xingwu1) [inline azext/batch/operations/job_operations.py]: why log here?
I think we should log it at cli extension by capturing the exception.

Comment (bgklein) [reply azext/batch/operations/job_operations.py]: Fair point.

**Pull request merged**